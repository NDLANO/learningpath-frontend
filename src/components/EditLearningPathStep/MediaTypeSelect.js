import React, { PropTypes } from 'react';
import Icon from '../Icon';
import polyglot from '../../i18n';

export default function MediaTypeSelect({value, onChange}) {
  const changeType = (evt) => onChange(evt.target.value);

  const fieldNameAttr = 'step_media_type';
  const fieldIdAttr = name => `${fieldNameAttr}_${name}`;

  return (
    <div className="icon-select">
      <div className="icon-select_item">
        <input
          type="radio" className="icon-select_input"
          id={fieldIdAttr('introduction')} name={fieldNameAttr}
          value="INTRODUCTION" checked={value === 'INTRODUCTION'} onChange={changeType}
        />
        <label htmlFor={fieldIdAttr('introduction')} className="icon-select_label">
          <Icon.TypeText />
          <span className="icon-select_label-text">{polyglot.t('editPathStep.mediatype.introduction')}</span>
        </label>
      </div>
      <div className="icon-select_item">
        <input
          type="radio" className="icon-select_input"
          id={fieldIdAttr('text')} name={fieldNameAttr}
          value="TEXT" checked={value === 'TEXT'} onChange={changeType}
        />
        <label htmlFor={fieldIdAttr('text')} className="icon-select_label">
          <Icon.TypeText />
          <span className="icon-select_label-text">{polyglot.t('editPathStep.mediatype.text')}</span>
        </label>
      </div>
      <div className="icon-select_item">
        <input
          type="radio" className="icon-select_input"
          id={fieldIdAttr('video')} name={fieldNameAttr}
          value="MULTIMEDIA" checked={value === 'MULTIMEDIA'} onChange={changeType}
        />
        <label htmlFor={fieldIdAttr('video')} className="icon-select_label">
          <Icon.TypeMedia />
          <span className="icon-select_label-text">{polyglot.t('editPathStep.mediatype.multimedia')}</span>
        </label>
      </div>
      <div className="icon-select_item">
        <input
          type="radio" className="icon-select_input"
          id={fieldIdAttr('quiz')} name={fieldNameAttr}
          value="QUIZ" checked={value === 'QUIZ'} onChange={changeType}
        />
        <label htmlFor={fieldIdAttr('quiz')} className="icon-select_label">
          <Icon.TypeQuiz />
          <span className="icon-select_label-text">{polyglot.t('editPathStep.mediatype.quiz')}</span>
        </label>
      </div>
      <div className="icon-select_item">
        <input
          type="radio" className="icon-select_input"
          id={fieldIdAttr('task')} name={fieldNameAttr}
          value="TASK" checked={value === 'TASK'} onChange={changeType}
        />
        <label htmlFor={fieldIdAttr('task')} className="icon-select_label">
          <Icon.TypeTask />
          <span className="icon-select_label-text">{polyglot.t('editPathStep.mediatype.task')}</span>
        </label>
      </div>
      <div className="icon-select_item">
        <input
          type="radio" className="icon-select_input"
          id={fieldIdAttr('summary')} name={fieldNameAttr}
          value="SUMMARY" checked={value === 'SUMMARY'} onChange={changeType}
        />
        <label htmlFor={fieldIdAttr('summary')} className="icon-select_label">
          <Icon.TypeSummary />
          <span className="icon-select_label-text">{polyglot.t('editPathStep.mediatype.summary')}</span>
        </label>
      </div>
    </div>
  );
}

MediaTypeSelect.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string
};
