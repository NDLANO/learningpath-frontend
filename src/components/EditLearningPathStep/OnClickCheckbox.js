import React, { PropTypes } from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import Tooltip from '../../common/tooltip/Tooltip';
import OverlayTrigger from '../../common/tooltip/OverlayTrigger';

const OnClickCheckbox = (field) => {
  const classes = {
    'learning-step-form_icon-bg': true,
    'learning-step-form_show-title': true,
    'learning-step-form_show-title--active': field.value
  };
  const tooltip = <Tooltip id="showTitleTooltip">Velg om titelen skal vises eller skjules</Tooltip>;

  return (
    <OverlayTrigger placement="top" overlay={tooltip}>
      <span className={classNames(classes)} onClick={() => field.onChange(!field.value)}>
        {field.value ? <Icon.Visibility /> : <Icon.VisibilityOff />}
        <label className="sr-only">
          <input type="checkbox" {...field} />
        </label>
      </span>
    </OverlayTrigger>
  );
};

OnClickCheckbox.propTypes = {
  field: PropTypes.object// .isRequired
};
export default OnClickCheckbox;
