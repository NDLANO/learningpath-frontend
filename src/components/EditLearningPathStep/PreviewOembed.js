import React, {PropTypes} from 'react';
import classNames from 'classnames';
import polyglot from '../../i18n';
import Oembed from '../Oembed';

export default class PreviewOembed extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      previewOembed: false
    };
  }

  onPreviewClick(evt) {
    this.setState({previewOembed: true});
    evt.preventDefault();
  }

  componentWillReceiveProps() {
    this.setState({previewOembed: false});
  }

  render() {
    const onPreviewClick = this.onPreviewClick.bind(this);
    const {content} = this.props;
    if (!content) {
      return null;
    }

    let frameClasses = classNames({
      'learningsource__frame': true,
      'learningsource__frame--hidden': !this.state.previewOembed
    });

    let previewButton = !this.state.previewOembed
    ? (<a className='learningsource__expand' onClick={onPreviewClick}>
      <span className='button button--outline'>{polyglot.t('editPathStep.previewOembed')}</span>
      </a>)
    : null;

    return (<div className='learningsource--wrapper'>
      <div className={frameClasses}>
        <Oembed oembedContent={content}/>
        {previewButton}
      </div>
    </div>);
  }
}

PreviewOembed.propTypes = {
  content: PropTypes.object.isRequired
};
