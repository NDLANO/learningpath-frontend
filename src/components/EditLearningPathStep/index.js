import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import assign from 'lodash/assign';
import { titleI18N, descriptionI18N, embedUrlI18N } from '../../util/i18nFieldFinder';
import LabeledIcon from '../LabeledIcon';
import TitleEditor from '../editors/TitleEditor';
import DescriptionHTMLEditor from '../editors/DescriptionHTMLEditor';
import polyglot from '../../i18n';
import MediaTypeSelect from './MediaTypeSelect';
import {
  updateLearningPathStep,
  updateLearningPathStepDescription,
  updateLearningPathStepTitle,
  updateLearningPathStepType,
  updateLearningPathStepEmbedUrl,
  validateOembed
} from '../../actions';

export function EditLearningPathStep (props, {lang}) {
  const {
    step,
    updateTitle,
    updateType,
    updateEmbedUrl,
    updateDescription,
    learningPathId,
    validateOembedUrl,
    saveAction,
    oembedIsValid
  } = props;
  const isValid = () => true;

  //const isOembedValid = isOembedValid === undefined ? true : step.isOembedValid;

  let saveLearningStep = () => saveAction(learningPathId, step);
  let title = titleI18N(step, lang) || '';
  let htmlDescription = descriptionI18N(step, lang) || '';
  let embedContent = embedUrlI18N(step, lang);

  let embedSourceInput = '';
  if (step.type) {
    embedSourceInput =(
      <div className='learningsource-form'>
        <div>
          <label className='mediatype-menu__label'>{polyglot.t('editPathStep.urlLabel')}</label>
          <input type='url' value={embedContent} onBlur={() => validateOembedUrl(embedContent)}
              onChange={(evt) => updateEmbedUrl({ url: evt.target.value, language: lang })}
              placeholder={polyglot.t('editPathStep.urlPlaceholder')} />
        </div>
      </div>
    );
  }

  return (
    <div className='learning-path-step'>
      <div className='learning-path_hd'>
        <h1 className='learing-path_title'>
          <TitleEditor lang={lang} value={title} onChange={updateTitle} />
        </h1>
      </div>
      <div className='learning-path_bd'>
        <div>
          <DescriptionHTMLEditor
            lang={lang}
            value={htmlDescription}
            onChange={updateDescription}
          />
        </div>

        <div className='mediatype-wrapper'>
          <MediaTypeSelect value={step.type} onChange={updateType} />

          {embedSourceInput}
        </div>
        <div>
          <button className='cta cta-link' onClick={saveLearningStep} disabled={ !isValid() || !oembedIsValid }>
            <LabeledIcon.Save labelText={polyglot.t('editPage.savePathBtn')} />
          </button>
        </div>
      </div>
    </div>
  );
}

EditLearningPathStep.propTypes = {
  step: PropTypes.object.isRequired,
  updateTitle: PropTypes.func.isRequired,
  updateType: PropTypes.func.isRequired,
  updateEmbedUrl: PropTypes.func.isRequired,
  updateDescription: PropTypes.func.isRequired,
  learningPathId: PropTypes.number.isRequired,
  saveAction: PropTypes.func.isRequired,
  oembedIsValid: PropTypes.bool.isRequired,
  validateOembedUrl: PropTypes.func.isRequired
};

EditLearningPathStep.contextTypes = {
  lang: PropTypes.string.isRequired
};

export const mapStateToProps = state => assign({}, state, {
  step: state.learningPathStep,
  learningPathId: state.learningPath.id

});

export const mapDispatchToProps = {
  // actions som endrer learningPathStep i redux store:
  updateTitle: updateLearningPathStepTitle,
  updateType: updateLearningPathStepType,
  updateEmbedUrl: updateLearningPathStepEmbedUrl,
  updateDescription: updateLearningPathStepDescription,
  // action til persistere learningPathStep
  saveAction: (learningPathId, lps) => updateLearningPathStep(learningPathId, lps.id, lps),

  validateOembedUrl: (embedContent) => validateOembed(embedContent)
};

export default connect(mapStateToProps, mapDispatchToProps)(EditLearningPathStep);
