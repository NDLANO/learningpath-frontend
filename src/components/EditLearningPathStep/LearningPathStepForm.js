import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import defined from 'defined';
import { reduxForm } from 'redux-form';
import { titleI18N, descriptionI18N, oembedUrlI18N, oembedContentI18N } from '../../util/i18nFieldFinder';
import { createValidator, required, oneOfIsRequired } from '../../util/validation';
import LabeledIcon from '../LabeledIcon';
import DescriptionHTMLEditor from '../editors/DescriptionHTMLEditor';
import MediaTypeSelect from './MediaTypeSelect';
import polyglot from '../../i18n';
import Icon from '../Icon';
import OnClickCheckbox from './OnClickCheckbox';
import OneLineEditor from '../editors/OneLineEditor';

import PreviewOembed from './PreviewOembed';
import LearningPathStepIcon from '../LearningPathStepIcon';
import {
  validateOembed,
} from '../../actions';

const LearningPathStepForm = (props) => {
  const {
    step,
    lang,
    oembedPreview,
    error,
    fields: { title, description, type, url, showTitle },
    handleSubmit,
    submitting,
    learningPathId,
  } = props;

  const embedContent = oembedContentI18N({embedContent: oembedPreview}, lang);

  const abortUrl = step.id ? `/learningpaths/${learningPathId}/step/${step.id}` : `/learningpaths/${learningPathId}`;

  if (!type.value) {
    return <MediaTypeSelect {...type} />;
  }

  return (
    <form onSubmit={handleSubmit} className="learning-step-form">
      <div className="learning-step-form_group">
        <span className="learning-step-form_mediatype-icon">
          <LearningPathStepIcon learningPathStepType={type.value} isCircle={false} />
        </span>
        <select className="learning-step-form_mediatype-dd" {...type} >
          <option value="INTRODUCTION">{polyglot.t('editPathStep.mediatype.introduction')}</option>
          <option value="TEXT">{polyglot.t('editPathStep.mediatype.text')}</option>
          <option value="MULTIMEDIA">{polyglot.t('editPathStep.mediatype.multimedia')}</option>
          <option value="QUIZ">{polyglot.t('editPathStep.mediatype.quiz')}</option>
          <option value="TASK">{polyglot.t('editPathStep.mediatype.task')}</option>
          <option value="SUMMARY">{polyglot.t('editPathStep.mediatype.summary')}</option>
        </select>
      </div>
      <div className="learning-step-form_group">
        <div className="learning-step-form_left">
          <span className="learning-step-form_icon-bg"><Icon.Create /></span>
          <OnClickCheckbox {...showTitle} />
        </div>
        <div className="learning-step-form_right">
          <div className="learning-step-form_input learning-step-form_title">
            <OneLineEditor lang={lang} {...title} placeholder={polyglot.t('editPathStep.titlePlaceHolder')} />
          </div>
          {title.touched && title.error && <span className="error_message error_message--red">{title.error}</span>}
        </div>
      </div>
      <DescriptionHTMLEditor lang={lang} {...description} />
      <div className="learning-step-form_group">
        <div className="learningsource-form">
          <div>
            <label className="mediatype-menu__label">{polyglot.t('editPathStep.urlLabel')}</label>
            <input
              type="url"
              {...url}
              placeholder={polyglot.t('editPathStep.urlPlaceholder')}
            />
            {url.touched && url.error && <span className="error_message error_message--red">{url.error}</span>}
            <PreviewOembed content={embedContent} />
          </div>
        </div>
        {(url.touched || description.touched) && error && <span className="error_message error_message--red">{error}</span>}
        <div className="block-container_fixed block-container_fixed--bottom--right">
          <div className="button-group">
            <Link to={abortUrl} className="button button--secondary">
              <LabeledIcon.Clear labelText={polyglot.t('editPage.cancelBtn')} />
            </Link>
            <button disabled={submitting} className="button button--primary" type="submit">
              <LabeledIcon.Save labelText={polyglot.t('editPage.savePathBtn')} />
            </button>
          </div>
        </div>
      </div>
    </form>
  );
};

LearningPathStepForm.propTypes = {
  lang: PropTypes.string.isRequired,
  error: PropTypes.string,
  fields: PropTypes.object.isRequired,
  step: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  learningPathId: PropTypes.number.isRequired,
  oembedPreview: PropTypes.array.isRequired,
  validateOembedUrl: PropTypes.func.isRequired
};


const mapStateToProps = (state, props) => ({
  oembedPreview: state.oembedPreview.oembedContent,
  initialValues: {
    showTitle: defined(props.step.showTitle, false),
    title: titleI18N(props.step, props.lang),
    description: descriptionI18N(props.step, props.lang),
    url: oembedUrlI18N(props.step, props.lang),
    type: props.step.type
  }
});

const mapDispatchToProps = {
  validateOembedUrl: (embedContent, lang) => validateOembed(embedContent, lang)
};


const asyncValidate = (values, dispatch, props) => {
  const { validateOembedUrl, lang } = props;
  return validateOembedUrl(values.url, lang);
};

const validate = createValidator({
  title: required(),
  _error: oneOfIsRequired('editPathStep.validation.oneOfDescriptionOrUrlIsRequired', 'url', 'description')
});

export default reduxForm({
  form: 'learning-path-step',
  fields: ['title', 'description', 'url', 'type', 'showTitle'],
  asyncValidate,
  validate,
  asyncBlurFields: ['url'],
}, mapStateToProps, mapDispatchToProps)(LearningPathStepForm);
