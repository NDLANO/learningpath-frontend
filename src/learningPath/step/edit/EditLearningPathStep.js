/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import assign from 'lodash/assign';
import get from 'lodash/get';

import LearningPathStepForm from './LearningPathStepForm';
import {
  fetchLearningPathStep,
  updateLearningPathStep,
  createLearningPathStep,
} from '../learningPathStepActions';
import { fetchLearningPathLicensesIfNeeded } from '../../edit/copyright/learningPathLicensesActions';
import polyglot from '../../../i18n';
import { getLearningPathStep } from '../learningPathStepSelectors';
import { getLearningPath } from '../../learningPathSelectors';
import { formattedEmbedUrl, formattedEmbedDescription, formattedEmbedLicense } from '../../../util/formatFormFieldsUtil';

export const saveStepObject = (step, values, language) => Object.assign({}, step, {
  type: values.type,
  showTitle: values.showTitle,
  title: values.title,
  language,
  description: formattedEmbedDescription(step, values.description),
  embedUrl: formattedEmbedUrl(step, values.url),
  license: formattedEmbedLicense(values.license),
});

class EditLearningPathStep extends Component {

  componentWillMount() {
    const { authenticated, localIfAuthenticated, localCreateEmptyLearningPathStep, fetchLearningPathLicenses, localFetchLearningPathStep, match: { params: { pathId, stepId } } } = this.props;
    fetchLearningPathLicenses('4.0');


    if (localIfAuthenticated && localCreateEmptyLearningPathStep) {
      localIfAuthenticated(authenticated, localCreateEmptyLearningPathStep);
    } else if (stepId) {
      localFetchLearningPathStep(pathId, stepId, true);
    }
  }

  render() {
    const { lang: language } = this.context;
    const {
      step,
      saveLearningPathStep,
      learningPath,
      licenses,
    } = this.props;
    if (!learningPath || !learningPath.id) {
      return null;
    }

    const handleSubmit = (values) => {
      const toSave = saveStepObject(step, values, language);
      return saveLearningPathStep(learningPath.id, toSave);
    };

    if (this.props.match.path === '/learningpaths/:pathId/step/new' && step.id) {
      return null;
    }

    return (
      <div className="two-column_content--wide learning-path-step two-column_content--white-bg">
        <LearningPathStepForm
          step={step}
          learningPath={learningPath}
          onSubmit={handleSubmit}
          lang={language}
          licenseOptions={licenses}
        />
      </div>
    );
  }
}

EditLearningPathStep.propTypes = {
  step: PropTypes.object.isRequired,
  saveLearningPathStep: PropTypes.func.isRequired,
  match: PropTypes.shape({
    path: PropTypes.string.isRequired,
    params: PropTypes.shape({
      pathId: PropTypes.string.isRequired,
      stepId: PropTypes.string,
    }).isRequired,
  }).isRequired,
  learningPath: PropTypes.object,
  fetchLearningPathLicenses: PropTypes.func.isRequired,
  localFetchLearningPathStep: PropTypes.func.isRequired,
  licenses: PropTypes.array.isRequired,
  localIfAuthenticated: PropTypes.func,
  localCreateEmptyLearningPathStep: PropTypes.func,
  authenticated: PropTypes.bool.isRequired,
};

EditLearningPathStep.contextTypes = {
  lang: PropTypes.string.isRequired,
};

export const mapStateToProps = state => assign({}, state, {
  step: getLearningPathStep(state),
  learningPath: getLearningPath(state),
  licenses: [{ description: polyglot.t('editPathStep.noLicenseChosen'), license: '' }].concat(get(state, 'learningPathLicenses.creativeCommonLicenses.all', [])),
});

export const mapDispatchToProps = {
  saveLearningPathStep: (learningPathId, lps) => (lps.id ? updateLearningPathStep(learningPathId, lps.id, lps) : createLearningPathStep(learningPathId, lps)),
  fetchLearningPathLicenses: fetchLearningPathLicensesIfNeeded,
  localFetchLearningPathStep: fetchLearningPathStep,
};

export default connect(mapStateToProps, mapDispatchToProps)(EditLearningPathStep);
