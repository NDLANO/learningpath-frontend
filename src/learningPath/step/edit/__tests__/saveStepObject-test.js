/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { saveStepObject } from "../EditLearningPathStep";

const stepWithId = {
  id: 1,
  title: "Old title",
  description: "Old description",
  someOtherField: "test",
};

const stepWithoutId = {
  title: "No id title",
  description: "No id description",
  someOtherField: "test",
};

test("util/saveStepObject description and embedUrl with step id", () => {
  const values = {
    title: "Test",
    description: "test",
    license: {
      license: "by-sa",
      description: "test",
    },
    showTitle: false,
    url: {
      embedType: "oembed",
      url: "https://ndla.no/id/3",
    },
  };

  expect(typeof saveStepObject).toBe("function");
  expect(saveStepObject(stepWithId, values, "nb")).toMatchSnapshot();
});

test("util/saveStepObject only description with step id", () => {
  const values = {
    title: "Test",
    description: "test",
    license: {
      license: "by-sa",
      description: "test",
    },
    showTitle: false,
    url: {
      embedType: "oembed",
      url: "",
    },
  };

  expect(typeof saveStepObject).toBe("function");
  expect(saveStepObject(stepWithId, values, "nb")).toMatchSnapshot();
});

test("util/saveStepObject only url with step id", () => {
  const values = {
    title: "Test",
    description: "",
    license: {
      license: "by-sa",
      description: "test",
    },
    showTitle: false,
    url: {
      embedType: "oembed",
      url: "https://ndla.no/id/3",
    },
  };

  expect(typeof saveStepObject).toBe("function");
  expect(saveStepObject(stepWithId, values, "nb")).toMatchSnapshot();
});

// formatFormFieldsUtil

test("util/saveStepObject description and embedUrl with no step id", () => {
  const values = {
    title: "Test",
    description: "test",
    license: {
      license: "by-sa",
      description: "test",
    },
    showTitle: false,
    url: {
      embedType: "oembed",
      url: "https://ndla.no/id/3",
    },
  };

  expect(typeof saveStepObject).toBe("function");
  expect(saveStepObject(stepWithoutId, values, "nb")).toMatchSnapshot();
});

test("util/saveStepObject only description with no step id", () => {
  const values = {
    title: "Test",
    description: "test",
    license: {
      license: "by-sa",
      description: "test",
    },
    showTitle: false,
    url: {
      embedType: "oembed",
      url: "",
    },
  };

  expect(typeof saveStepObject).toBe("function");
  expect(saveStepObject(stepWithoutId, values, "nb")).toMatchSnapshot();
});

test("util/saveStepObject only url with no step id", () => {
  const values = {
    title: "Test",
    description: "",
    license: {
      license: "by-sa",
      description: "test",
    },
    showTitle: false,
    url: {
      embedType: "oembed",
      url: "https://ndla.no/id/3",
    },
  };

  expect(typeof saveStepObject).toBe("function");
  expect(saveStepObject(stepWithoutId, values, "nb")).toMatchSnapshot();
});
