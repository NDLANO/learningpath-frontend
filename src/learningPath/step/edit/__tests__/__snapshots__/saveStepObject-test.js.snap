// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/saveStepObject description and embedUrl with no step id 1`] = `
Object {
  "description": "test",
  "embedUrl": Object {
    "embedType": "oembed",
    "url": "https://ndla.no/id/3",
  },
  "language": "nb",
  "license": "by-sa",
  "showTitle": false,
  "someOtherField": "test",
  "title": "Test",
  "type": undefined,
}
`;

exports[`util/saveStepObject description and embedUrl with step id 1`] = `
Object {
  "description": "test",
  "embedUrl": Object {
    "embedType": "oembed",
    "url": "https://ndla.no/id/3",
  },
  "id": 1,
  "language": "nb",
  "license": "by-sa",
  "showTitle": false,
  "someOtherField": "test",
  "title": "Test",
  "type": undefined,
}
`;

exports[`util/saveStepObject only description with no step id 1`] = `
Object {
  "description": "test",
  "embedUrl": undefined,
  "language": "nb",
  "license": "by-sa",
  "showTitle": false,
  "someOtherField": "test",
  "title": "Test",
  "type": undefined,
}
`;

exports[`util/saveStepObject only description with step id 1`] = `
Object {
  "description": "test",
  "embedUrl": Object {
    "embedType": "oembed",
    "url": "",
  },
  "id": 1,
  "language": "nb",
  "license": "by-sa",
  "showTitle": false,
  "someOtherField": "test",
  "title": "Test",
  "type": undefined,
}
`;

exports[`util/saveStepObject only url with no step id 1`] = `
Object {
  "description": undefined,
  "embedUrl": Object {
    "embedType": "oembed",
    "url": "https://ndla.no/id/3",
  },
  "language": "nb",
  "license": "by-sa",
  "showTitle": false,
  "someOtherField": "test",
  "title": "Test",
  "type": undefined,
}
`;

exports[`util/saveStepObject only url with step id 1`] = `
Object {
  "description": "",
  "embedUrl": Object {
    "embedType": "oembed",
    "url": "https://ndla.no/id/3",
  },
  "id": 1,
  "language": "nb",
  "license": "by-sa",
  "showTitle": false,
  "someOtherField": "test",
  "title": "Test",
  "type": undefined,
}
`;
